rero_ils/modules/cli.py

```diff
@@ -475,3 +477,57 @@ def compile_json(src_jsonfile, output, verbose):
     if not output:
         output = sys.stdout
     json.dump(data, fp=output, indent=2)
+
+
+@utils.command('reserve_pid_range')
+@click.option('-p', '--pid_type', 'pid_type', default=None,
+              help='pid type of the resource')
+@click.option('-n', '--records_number', 'records_number', default=None,
+              help='Number of records to load')
+@click.option('-u', '--unused', 'unused', is_flag=True, default=False,
+              help='Set unused pids status to NEW ')
+@with_appcontext
+def reserve_pid_range(pid_type, records_number, unused):
+    """Reserve a range of pids for eventual records loading.
+
+    reserved pids will have the status RESERVED.
+    - pid_type: the pid type of the resource as configured in config.py
+    - records_number: number of new records(with pids) to load.
+    - unused: set that the status of unused pids to NEW.
+    """
+    click.secho('Reserving pids for loading "%s" records' %
+                pid_type, fg='green')
+    try:
+        records_number = int(records_number)
+    except ValueError:
+        raise ValueError('Parameter records_number must be integer.')
+
+    try:
+        record_class = obj_or_import_string(
+            current_app.config
+            .get('RECORDS_REST_ENDPOINTS')
+            .get(pid_type).get('record_class', Record))
+    except AttributeError:
+        raise AttributeError('Invalid pid type.')
+
+    identifier = record_class.provider.identifier
+    reserved_pids = []
+    for number in range(0, records_number):
+        pid = identifier.next()
+        reserved_pids.append(pid)
+        record_class.provider.create(pid_type, pid_value=pid,
+                                     status=PIDStatus.RESERVED)
+        db.session.commit()
+    click.secho(
+        ('reserved_pids range, from: {min} to: {max}').format(
+            min=min(reserved_pids), max=max(reserved_pids)
+        ))
+    if unused:
+        for pid in range(1, identifier.max()):
+            if not db.session.query(
+                    identifier.query.filter(identifier.recid == pid).exists()
+            ).scalar():
+                record_class.provider.create(pid_type, pid_value=pid,
+                                             status=PIDStatus.NEW)
+                db.session.add(identifier(recid=pid))
+            db.session.commit()
```

Comment:
Despite I tried to used "-u" option, all requested pid are set to RESERVED into PostgreSQL.
I tried with multiple record_type, multiple record_number.
My Invenio knowledge is not suffisant to determine if it's a bug or not.... But for my tests the `if` condition is never satisfied.

Reply:
If you have the following scenario in db:
1. REGISTERED pids 1,2,3,7,8,9
2. you execute the cli to reserve 10 pids, with the option `-u`
3. here is what you get:
3.1: REGISTERED pids 1,2,3,7,8,9 [pid_status in table = 'R']
3.2: NEW pids: 4,5,6, [pid_status in table = 'N']
3.3. RESERVED pids: 10->19  [pid_status in table = 'K']

so the option `u` is only to fill the gaps.